fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java

```diff
@@ -73,7 +74,15 @@ void setStreamId(String streamId) {
     private final ObjectMapper objectMapper;
     private final Map<String, SubscriptionStream> streams;
 
+    public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, AccessTokenProvider accessTokenProvider) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, accessTokenProvider), true);
+    }
+
     public ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory) {
+        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, null), true);
+    }
+
+    ManagedCursorManager(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, boolean clientHttpRequestFactoryIsAlreadyWrapped) {
```

Comment:
I would have to check for both `instanceof` `AuthorizedClientHttpRequestFactory` and `ProblemHandlingClientHttpRequestFactory`, the authorization wrapping is only done if there is a ´AccessTokenProvider`.